### ISSUE TITLE
Virtual text isn't cleared 'on demand'

### ISSUE THREAD

**Issue by igorlfs** [theHamsta/nvim-dap-virtual-text#74]
Hey!

Thanks for the nice plugin :)

I'm having trouble hiding the virtual text. There is some discussion about this issue in another thread:

              Even `:DapVirualTextDisable` doesn't clear the virt text for me: no new text is produced, but the old text does not hide.

_Originally posted by @crhf in https://github.com/theHamsta/nvim-dap-virtual-text/issues/14#issuecomment-1625096010_

From what I understand, reading the original discussion, is that the logic to hide the virtual text is coupled with nvim-dap's event system. Nonetheless, it would be nice if commands like `:DapVirtualTextDisable` and `:DapVirtualTextToggle` could actually hide the text.

**Comment by nishantpillai5** [comment]
+1

**Comment by TheLeoP** [comment]
You can add a command to do this:

```lua
vim.api.nvim_create_user_command(
  "DapVirtualTextClear",
  function() require("nvim-dap-virtual-text.virtual_text").clear_virtual_text() end,
  {
    desc = "Clear all the virtual text displayed by nvim-dap-virtual-text",
  }
)
```

**Comment by igorlfs** [comment]
Hey, @TheLeoP, thanks for the suggestion!

I decided to take a deeper look and noticed that reverting commit 57f1dbd fixes the issue. This commit should reduce flickering, but with my (very) limited testing, I haven't noticed any flickering.

**Comment by theHamsta** [comment]
I assume this is fixed by #78?

**Comment by igorlfs** [comment]
Yep
